specs/composition-full/paths/rules/rule.yml

```
@@ -27,3 +27,67 @@ get:
       $ref: '../../../common/responses/MethodNotAllowed.yml'
     '404':
       $ref: '../../../common/responses/IndexNotFound.yml'
+
+put:
+  tags:
+    - Rules
+  operationId: putCompositionRule
+  x-acl:
+    - editSettings
+    - settings
+  summary: Upsert a composition rule
+  description: |
+    Upsert a Composition Rule for the specified composition ID.
+  parameters:
+    - $ref: '../urlParams.yml#/compositionID'
+    - $ref: './parameters.yml#/ObjectIDRule'
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          $ref: '../../common/schemas/components/CompositionRule.yml#/compositionRule'
+  responses:
+    '200':
+      description: OK
+      content:
+        application/json:
+          schema:
+            $ref: '../../../common/responses/common.yml#/taskID'
+    '400':
+      $ref: '../../../common/responses/BadRequest.yml'
+    '402':
+      $ref: '../../../common/responses/FeatureNotEnabled.yml'
+    '403':
+      $ref: '../../../common/responses/MethodNotAllowed.yml'
+    '422':
+      $ref: '../../../common/responses/UnprocessableEntity.yml'
+
+delete:
+  tags:
+    - Rules
+  operationId: deleteCompositionRule
+  x-acl:
+    - editSettings
+    - settings
+  summary: Delete a Composition Rule
+  description: |
+    Delete a Composition Rule from the specified Composition ID.
+  parameters:
+    - $ref: '../urlParams.yml#/compositionID'
+    - $ref: './parameters.yml#/ObjectIDRule'
+  responses:
+    '200':
+      description: OK
+      content:
+        application/json:
+          schema:
+            $ref: '../../../common/responses/common.yml#/taskID'
+    '400':
+      $ref: '../../../common/responses/BadRequest.yml'
+    '402':
+      $ref: '../../../common/responses/FeatureNotEnabled.yml'
+    '403':
+      $ref: '../../../common/responses/MethodNotAllowed.yml'
+    '404':
+      $ref: '../../../common/responses/CompositionNotFound.yml'
```

```suggestion
      oneOf:
            - $ref: '../../../common/responses/CompositionNotFound.yml'
            - $ref: '../../../common/responses/CompositionRuleNotFound.yml'
```
The second probably needs added

---

I don't think we ever respond with CompositionRuleNotFound, we always return a taskID, even if the Comp Rule doesn't exist.